python/interpret_community/common/model_wrapper.py

```diff
@@ -344,6 +370,14 @@ def _classifier_without_proba(model):
     return isinstance(model, SGDClassifier) and not hasattr(model, SKLearn.PREDICT_PROBA)
 
 
+def _does_predict_gives_dataframe(model, examples):
+    import pandas as pd
+    if isinstance(model.predict(examples.dataset), pd.DataFrame) or \
```

Comment:
calling predict on full dataset is a bit wasteful, I think this can either be added in _eval_function method, after:

result = function(examples.typed_wrapper_func(examples_dataset[0]))

or we can keep this and just use something similar to _eval_model and only select first instance:

result = model.predict_proba(examples.typed_wrapper_func(examples.dataset[0:1]))

I think either approach works fine, maybe second one is easier and also better to make

Reply:
Moved the logic inside the classification and regression wrapped classes. Please have a look.